jupyterhub/scopes.py

```diff
@@ -6,33 +6,36 @@
 from tornado.log import app_log
 
 from . import orm
+from . import roles
 
 
 class Scope(Enum):
     ALL = True
 
 
-def get_user_scopes(name):
-    """
-    Scopes have a metascope 'all' that should be expanded to everything a user can do.
-    At the moment that is a user-filtered version (optional read) access to
-    users
-    users:name
-    users:groups
-    users:activity
-    users:servers
-    users:tokens
-    """
-    scope_list = [
-        'users',
-        'users:name',
-        'users:groups',
-        'users:activity',
-        'users:servers',
-        'users:tokens',
-    ]
-    scope_list.extend(['read:' + scope for scope in scope_list])
-    return {"{}!user={}".format(scope, name) for scope in scope_list}
+def get_scopes_for(orm_object):
+    """Find scopes for a given user or token"""
+    scopes = set()
+    if orm_object is None:
+        return scopes
+    elif isinstance(orm_object, orm.APIToken):
+        owner = orm_object.user or orm_object.service
+        token_scopes = roles.get_scopes_for(orm_object)
```

Comment:
I think it's a little confusing to have two `get_scopes_for` that do different things. What do the do differently, and can we perhaps get a clearer name for one or the other?

Reply:
Good point. I renamed one of them